backend/src/api/routes/moving_averages.py

```
@@ -0,0 +1,345 @@
+        # Calculate moving average
+        service = MovingAverageService()
+        if ma_type == 'sma':
+            ma_values = service.calculate_sma(series, window)
+        elif ma_type == 'ema':
+            ma_values = service.calculate_ema(series, window)
+        elif ma_type == 'wma':
+            if not weights:
+                raise HTTPException(
+                    status_code=400,
+                    detail="'weights' required for WMA"
+                )
```

Ruff (0.14.4)

Abstract `raise` to an inner function

(TRY301)